grudge/trace_pair.py

```diff
@@ -377,54 +377,40 @@ def finish(self):
 
 
 class _RankBoundaryCommunicationLazy:
+    base_tag = 1273
 
-    def __init__(self, dcoll: DiscretizationCollection, remote_rank,
-                 vol_field, tag):
+    def __init__(self,
```

Comment:
@inducer/@thomasgibson: Could you please take a brief look if this commit seems reasonable? In particular, the flatten/unflatten stuff and the tag handling.